### ISSUE TITLE
Warning: "/bin/sh: git: command not found"

### ISSUE THREAD

**Issue by dliessi** [openlilylib/snippets#49]
When I compile a file including https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/definitions.ily and using some Git commands (e.g. `\gitAuthor`) I get some warnings saying "/bin/sh: git: command not found".
If I understand correctly, that file expects Git to be in `$PATH`, but this is not always the case.

I think that the location of external tools used by openlilylib should be customizable via a configuration file.
The configuration file should not be tracked by Git, otherwise it would be a problem when updating the repository.

**Comment by uliska** [comment]
I knew this only works on Linux but didn't find a viable solution so far 00

On 30. August 2014 12:23:47 MESZ, Davide Liessi notifications@github.com wrote:

> When I compile a file including
> https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/definitions.ily
> and using some Git commands (e.g. `\gitAuthor`) I get some warnings
> saying "/bin/sh: git: command not found".
> If I understand correctly, that file expects Git to be in `$PATH`, but
> this is not always the case.
> 
> I think that the location of external tools used by openlilylib should
> be customizable via a configuration file.
> The configuration file should not be tracked by Git, otherwise it would
> be a problem when updating the repository.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/openlilylib/openlilylib/issues/49

## 

Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by uliska** [comment]
I knew this only works on Linux but didn't find a viable solution so far (no idea about Windows, though).

Does this make a problem beyond a scrambled tagline in your case?

On 30. August 2014 12:23:47 MESZ, Davide Liessi notifications@github.com wrote:

> When I compile a file including
> https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/definitions.ily
> and using some Git commands (e.g. `\gitAuthor`) I get some warnings
> saying "/bin/sh: git: command not found".
> If I understand correctly, that file expects Git to be in `$PATH`, but
> this is not always the case.
> 
> I think that the location of external tools used by openlilylib should
> be customizable via a configuration file.
> The configuration file should not be tracked by Git, otherwise it would
> be a problem when updating the repository.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/openlilylib/openlilylib/issues/49

## 

## Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by dliessi** [comment]
No, the only problem is the tagline, i.e. the Git commands seem to return an empty string.

I was thinking of something like an optional configuration file containing e.g.

```
gitexecutable="/opt/local/bin/git"
```

In https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/definitions.ily you would define `gitexecutable="git"`, then check if the configuration file exists and in that case import it, and then use `gitexecutable` in `\gitCommand` instead of the hardcoded `"git"`.

The configuration file name/path should be added to `.gitignore`.

I'll try to prepare a patch for this in the next days, if you agree.

Is Git the only external tool invoked by openlilylib?
What would be an appropriate name and location for the configuration file?
(If Git is the only tool, maybe the file could be placed in https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/)

**Comment by jan-warchol** [comment]
2014-08-30 16:00 GMT+02:00 Davide Liessi notifications@github.com:

> No, the only problem is the tagline, i.e. the Git commands seem to return
> an empty string.
> 
> I was thinking of something like an optional configuration file containing
> e.g.
> 
> gitexecutable="/opt/local/bin/git"
> 
> In
> https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/definitions.ily
> you would define gitexecutable="git", then check if the configuration
> file exists and in that case import it, and then use gitexecutable in
> \gitCommand instead of the hardcoded "git".
> 
> The configuration file name/path should be added to .gitignore.
> 
> I'll try to prepare a patch for this in the next days, if you agree.
> 
> Is Git the only external tool invoked by openlilylib?
> 
> I think so.
> 
> What would be an appropriate name and location for the configuration file?
> (If Git is the only tool, maybe the file could be placed in
> https://github.com/openlilylib/openlilylib/blob/master/general-tools/git-commands/
> )
> 
> That should be fine.

**Comment by uliska** [comment]
> No, the only problem is the tagline, i.e. the Git commands seem to return an empty string. I was thinking of something like an optional configuration file containing e.g. `gitexecutable="/opt/local/bin/git"`

@dliessi To come back to this, what would `which git` return on Mac?

**Comment by dliessi** [comment]
> To come back to this, what would `which git` return on Mac?

By default nothing, since Git is not installed by the system; with MacPorts generally `/opt/local/bin/git`; with Homebrew (and probably also installing from source or with the prebuilt binary) `/usr/local/bin/git`.

Anyway this is only if the `$PATH` is configured correctly, but on Mac graphic applications don't inherit the `$PATH` from the user: they use instead a system wide `$PATH` (that by default is empty, if I remember correctly) and it is extremely unlikely that the user touched it.

**Comment by dliessi** [comment]
> I'll try to prepare a patch for this in the next days

Oops! I didn't remember of this issue.
I'll put it on the todo list, if you want.

**Comment by uliska** [comment]
Well, if _you_ want. But I'd prefer a nice audio file ;-)

**Comment by dliessi** [comment]
> But I'd prefer a nice audio file ;-)

:D
